### ISSUE TITLE
Doesn’t play well with `-Ywarn-unused-import`.

### ISSUE THREAD

**Issue by sellout** [lihaoyi/acyclic#16]
We use `-Ywarn-unused-import` (and `-Xfatal-warnings`), and get
```
Unused import
import acyclic.file
```
on any file I add it to.

I don’t know of any way to silence an unused import warning (e.g., using `@SuppressWarnings`), but I’d happily do that if possible.

**Comment by rintcius** [comment]
@sellout maybe this plugin can help? https://github.com/ghik/silencer

**Comment by lihaoyi** [comment]
One solution would be to make the acyclic plugin remove that import when it sees it. If anyone wants to send a PR doing so, I'll gladly merge this

**Comment by lihaoyi** [comment]
Or you could use `-P:acyclic:force` which would remove the need for those imports in the first place, assuming your entire codebase is already compliant

**Comment by sellout** [comment]
Yeah, our codebase is far from compliant. We’ll need to add the import one file at a time.

**Comment by Daenyth** [comment]
If the `acyclic` imports defined some `def workaroundUnusedImportWarning: Unit = ()` then would that solve the problem? File level imports could then
```scala
import acyclic.file

object MyClass {
  file.ignoreUnusedImport
```

**Comment by caoilte** [comment]
Adding `-P:acyclic:force` doesn't help if you want to prevent package level cycles (an awesome feature by the way). 

Silencer is fine for handling this, but I'm curious for why you didn't use annotations instead of an import? That feels like it would read better and be less surprising.

**Comment by cornerman** [comment]
If anyone still needs this, you can silence these warnings with the newish `Wconf` compiler option:

```
-Wconf:cat=unused-imports&origin=acyclic\.skipped:s
```
